### DISCUSSION TITLE
Allow custom extensions to access system context during initialization

### DISCUSSION THREAD

**Discussion by maartenvn** [directus/directus#3032] [Feature Requests]
**Directus**: 9.0.0-rc.0
**Postgres**: 13

The app uses `defineInterface` when creating interfaces, which allows passing the `i18n` as context.
According to @rijkvanzanten custom interfaces should be able to just export a function that receives the `i18n` object as context.
This is not the case, as only objects are supported in the module loading code:

https://github.com/directus/next/blob/20355fee5eba514dd75565f60269311187010c66/app/src/interfaces/register.ts#L24

As you can see the `default` is used without calling a potential function with the correct context

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
